### ISSUE TITLE
CommandBar overflow menu callout is renderer incorrectly when small space provided

### ISSUE THREAD

**Issue by PatoBeltran** [microsoft/fluentui#18227]
### Environment Information

- **Package version(s)**:  fluent 8 latest

### Please provide a reproduction of the bug in a codepen:

https://codepen.io/PatoBeltran/pen/LYWRyNX?editors=0110

#### Actual behavior:

When the callout doesn't have a lot of space to render, it seems like the positioning logic doesn't correctly know how to render and it ends up either rendering the callout over the host button of rendering the beak incorrectly. To repro this open the codepen and go to 400% zoom and make sure your browser window is not overly huge and you will see the callout being positioned in weird places. Also notice that the beak doesn't appear at first, but when you're zoomed in it does appear.

#### Expected behavior:

The callout should never be positioned on top of the host button, or at the very least the callout shouldn't make the user think that the position is incorrect.

### Priorities and help requested:

Products/sites affected: Azure IoT Central

**Comment by gouttierre** [comment]
@PatoBeltran - thanks for reaching out to us with this issue. 

@miroslavstastny - Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @smhigley

**Comment by ling1726** [comment]
This is definitely a bug, on 400% the beak starts to render, however on 100% zoom there is no beak rendering
